### PR TITLE
docs: Add docker example

### PR DESCRIPTION
**Pull request by pavelzw**
Related to #844

### CODE REVIEW

**Comment by pavelzw** [comment]
Hmm, it might make sense to remove the other https://github.com/prefix-dev/pixi/tree/main/examples/docker-build example?

**Comment by ruben-arts** [comment]
> Hmm, it might make sense to remove the other [main/examples/docker-build](https://github.com/prefix-dev/pixi/tree/main/examples/docker-build?rgh-link-date=2024-02-24T13%3A35%3A13Z) example?

Yeah I think it does make sense!

File: examples/docker/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# docker
+
+This example is using docker in combination with [solve-groups](https://pixi.sh/latest/configuration/#the-environments-table).
+The solve-groups ensure that the `default` environment (where the tests are run) is using *exactly* the same versions of the dependencies as the `prod` environment.
+
+In the docker container, we only copy the `prod` environment into the final layer, so the `default` environment and all its dependencies are not included in the final image.
+Also, `pixi` itself is not included in the final image and we activate the environment using `pixi -e prod shell-hook`.
```
Comment (ruben-arts) [inline examples/docker/README.md:7]:
````suggestion
Also, `pixi` itself is not included in the final image and we activate the environment using `pixi -e prod shell-hook`.

## Usage
To build and run the docker container you require [`docker`](https://docs.docker.com/engine/install/)
When you have `docker` use the following commands:

```shell
docker build -t pixi-docker .
docker run -p 8000:8000 pixi-docker
```
````

I think adding a usage block would be helpful

**Review by ruben-arts — APPROVED** [review]

**Comment by ruben-arts** [comment]
Thanks pavel!

**Pull request merged**